invenio_rdm_records/services/review/service.py

```diff
@@ -143,18 +144,44 @@ def delete(self, identity, id_, revision_id=None, uow=None):
         return True
 
     @unit_of_work()
-    def submit(self, identity, id_, data=None, revision_id=None, uow=None):
-        """Submit record for review."""
-        # Get record and check permission
+    def submit(
+        self, identity, id_, data=None, direct_publish=None, revision_id=None, uow=None
+    ):
+        """Submit record for review or direct publish to the community."""
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
-        self.require_permission(identity, "update_draft", record=draft)
-
         # Preconditions
         if draft.parent.review is None:
             raise ReviewNotFoundError()
 
         # since it is submit review action, assume the receiver is community
         resolved_community = draft.parent.review.receiver.resolve()
+        if direct_publish:
+            # Direct publish to community workflow
+            current_communities.service.require_permission(
+                identity, "direct_publish", record=resolved_community
+            )
+
+            # Submit draft for review
+            submitted_review = self._submit(
+                identity, draft, resolved_community, data, revision_id, uow=uow
+            )
+            # Accept draft in community
+            request_item = current_requests_service.execute_action(
+                identity, submitted_review.data["id"], "accept", data=data, uow=uow
+            )
+            return request_item
```

Comment:
is the parent record already committed/indexed here?

Reply:
"No" but yes. The `_submit` method registers a `RecordCommitOp`. So it will be committed by the UoW and fail otherwise.